docs/reference/block_allocator.rst

```diff
@@ -0,0 +1,47 @@
+Block Allocator
+===============
+
+Previous Design
+---------------
+
+Historically there have been two image allocators in Pillow:
+``ImagingAllocateBlock`` and ``ImagingAllocateArray``. The first works
+for images smaller than 16MB of data and allocates one large chank of
```

Comment:
chunk?

Reply:
chank, chunk. clunk.  they all sound the same.